src/coreclr/src/System.Private.CoreLib/src/System/Threading/ClrThreadPoolBoundHandle.Windows.cs

```diff
@@ -16,6 +16,7 @@ private static ThreadPoolBoundHandle BindHandleCore(SafeHandle handle)
 
             try
             {
+                Debug.Assert(OperatingSystem.IsWindows());
```

Comment:
Why is this needed if the file .Windows.cs ?

Reply:
As mentioned [in my other commment](https://github.com/dotnet/runtime/pull/43097#discussion_r500433469) my effort of adding [SupportedOSPlatform("windows")] attribute in AssemblyInfo.cs of windows only build is worked only for libraries build, not for corelib build, seems they built differently and couldn't fund out how corelib windows (or any platform) specific build setup. It seems to me there is no windows specific build or so, so all warnings found in corelib are suppressed like this or using other suppressing ways

Reply:
> It seems to me there is no windows specific build or so

cc @ericstj @safern @ViktorHofer please correct me if am wrong and point me in the right direction

Reply:
corelib at least from coreclr defines `TargetsWindows/Unix/Browser`. So you could use those properties. See here: https://github.com/dotnet/runtime/blob/master/src/coreclr/dir.common.props#L55-L63

Reply:
>  It seems to me there is no windows specific build or so, so all warnings found in corelib are suppressed like this or using other suppressing ways

There are OS specific builds of CoreLib but it doesn't multi-target as we do in libraries. Meaning it uses `TargetOS` and other properties that are passed in via a root build.

Reply:
Also note that there are two versions of CoreLib, one in the coreclr directory and one in Mono.

Reply:
Thanks @safern @ViktorHofer i will give it try

Reply:
> Also note that there are two versions of CoreLib, one in the coreclr directory and one in Mono.

I have run mono build, didn't see any warning, but I am not sure if the Mono build is using the analyzer i have modified

Reply:
> corelib at least from coreclr defines TargetsWindows/Unix/Browser. So you could use those properties. See here:

@safern  I am actually using the same property (` <TargetsWindows Condition="'$(TargetOS)' == 'Windows_NT'">true</TargetsWindows> `) in `versionioning.targets` but it is not working for corelib, seems corelib build is not using (importing) `versionioning.targets` in its build, the question might be what else i can use instead of `versionioning.targets` 

> There are OS specific builds of CoreLib but it doesn't multi-target as we do in libraries. 

@ViktorHofer what i am trying to do is `adding [SupportedOSPlatform("windows")] attribute in AssemblyInfo.cs of windows only build` but i don't see windows specific output with `AssemblyInfo.cs` for CoreLib, i see it only for platform less build result. That means even there were `OS specific builds of CoreLib` if it is not producing `OS specific AssemblyInfo.cs` then this i cannot suppress the warnings for windows only build

Reply:
> seems corelib build is not using (importing) versionioning.targets in its build

correct, `versioning.targets` is a libraries specific msbuild file that isn't imported in CoreLib.

Reply:
> didn't see any warning, but I am not sure if the Mono build is using the analyzer

It should and if not it needs fixing

Reply:
> > didn't see any warning, but I am not sure if the Mono build is using the analyzer i have **modified**, or if they use same analyzer settings as in the https://github.com/dotnet/runtime/blob/master/eng/Analyzers.props#L9
> 
> It should and if not it needs fixing

I meant i am not sure if mono uses the same analyzer settings as in [libraries](https://github.com/dotnet/runtime/blob/master/eng/Analyzers.props#L9), therefore i am not sure if my changes affected mono build

Reply:
FYI Mono corelib is consuming the same analyzer versions, no warnings on the build, i can see the warnings if intentionally add platform-specific attributes

Reply:
I have added the attribute in the coreclr windows build and the build result looks good, warnings are gone. But when run libs build it used windows version of the coreclr build which having assembly level "Supported" on windows attribute (`runtime\artifacts\bin\coreclr\Windows_NT.x64.Debug\IL\System.Private.CoreLib.dll`) which would cause much more warnings. So seems for coreclr better not to add the attribute for targeted build